mesonbuild/interpreter.py

```diff
@@ -1214,6 +1213,97 @@ def get_cross_property_method(self, args, kwargs):
+                          'build_by_default',
+                          'vs_module_defs',
+                          'pic',
+                          'override_options',
+                      ])
+
+build_target_common_kwargs = set()
+build_target_common_kwargs.update(buildtarget_kwargs)
+build_target_common_kwargs.update(lang_arg_kwargs)
+build_target_common_kwargs.update(pch_kwargs)
+build_target_common_kwargs.update(vala_kwargs)
+build_target_common_kwargs.update(rust_kwargs)
+build_target_common_kwargs.update(cs_kwargs)
+
+exe_kwargs = set()
+exe_kwargs.update(build_target_common_kwargs)
+
+shlib_kwargs = set()
+shlib_kwargs.update(build_target_common_kwargs)
+shlib_kwargs.update(['version', 'soversion'])
+shmod_kwargs = shlib_kwargs
+stlib_kwargs = shlib_kwargs
+
+jar_kwargs = exe_kwargs.copy()
+jar_kwargs.update(['main_class'])
+
+build_target_kwargs = exe_kwargs.copy()
+build_target_kwargs.update(['target_type'])
+
+permitted_kwargs = {'project': set(['version', 'meson_version', 'default_options', 'license', 'subproject_dir']),
+                    'executable': exe_kwargs,
+                    'shared_library': shlib_kwargs,
+                    'static_library': stlib_kwargs,
+                    'shared_module': shmod_kwargs,
+                    'jar': jar_kwargs,
+                    'build_target': build_target_kwargs,
+                    'custom_target': set(['input', 'output', 'command', 'install', 'install_dir', 'build_always', 'capture', 'depends', 'depend_files', 'depfile', 'build_by_default']),
+                    'run_target': set(['command', 'depends']),
+                    'generator': set(['arguments', 'output', 'depfile']),
+                    'test': set(['args', 'env', 'is_parallel', 'should_fail', 'timeout', 'workdir', 'suite']),
+                    'benchmark': set(['args', 'env', 'should_fail', 'timeout', 'workdir', 'suite']),
+                    'install_headers': set(['install_dir', 'subdir']),
+                    'install_man': set(['install_dir']),
+                    'install_data': set(['install_dir', 'install_mode', 'sources']),
+                    'install_subdir': set(['install_dir', 'install_mode']),
+                    'configure_file': set(['input', 'output', 'configuration', 'command', 'install_dir', 'capture', 'install']),
+                    'include_directories': set(['is_system']),
+                    'add_global_arguments': set(['language']),
+                    'add_project_arguments': set(['language']),
+                    'add_test_setup': set(['exe_wrapper', 'gdb', 'timeout_multiplier', 'env']),
+                    'add_languages': set(['required']),
+                    'find_program': set(['required',]),
+                    'run_command': set(['in_builddir']), # INTERNAL
+                    'subproject': set(['version', 'default_options']),
+                    'vcs_tag': set(['input', 'output', 'fallback', 'command', 'replace_string']),
+}
```

Comment:
__[flake8]__
* [E124] closing bracket does not match visual indentation

<!-- Comment by [SideCI](https://sideci.com) -->